torchci/pages/api/drci/drci.ts

```diff
@@ -101,93 +101,97 @@ export async function updateDrciComments(
-      FLAKY: flakyJobs,
-      BROKEN_TRUNK: brokenTrunkJobs,
-      UNSTABLE: unstableJobs,
-    };
-
-    const failureInfo = constructResultsComment(
-      pending,
-      failedJobs,
-      flakyJobs,
-      brokenTrunkJobs,
-      unstableJobs,
-      pr_info.head_sha,
-      pr_info.merge_base,
-      pr_info.merge_base_date,
-      `${HUD_URL}${OWNER}/${repo}/${pr_info.pr_number}`
-    );
+      const comment = formDrciComment(
+        pr_info.pr_number,
+        OWNER,
+        repo,
+        failureInfo,
+        formDrciSevBody(sevs)
+      );
 
-    const comment = formDrciComment(
-      pr_info.pr_number,
-      OWNER,
-      repo,
-      failureInfo,
-      formDrciSevBody(sevs)
-    );
+      const { id, body } =
+        existingDrCiComments.get(pr_info.pr_number) ||
+        (await getDrciComment(octokit, OWNER, repo, pr_info.pr_number));
 
-    const { id, body } =
-      existingDrCiComments.get(pr_info.pr_number) ||
-      (await getDrciComment(octokit, OWNER, repo, pr_info.pr_number));
+      // The comment is there and remains unchanged, so there is no need to do anything
+      if (body === comment) {
+        return;
+      }
 
-    // The comment is there and remains unchanged, so there is no need to do anything
-    if (body === comment) {
-      return;
-    }
+      // If the id is 0, it means that the bot has failed to create the comment, so we
+      // are free to create a new one here
+      if (id === 0) {
+        await octokit.rest.issues.createComment({
+          body: comment,
+          owner: OWNER,
+          repo: repo,
+          issue_number: pr_info.pr_number,
+        });
+      }
+      // Otherwise, update the existing comment
+      else {
+        await octokit.rest.issues.updateComment({
+          body: comment,
+          owner: OWNER,
+          repo: repo,
+          comment_id: id,
+        });
+      }
 
-    // If the id is 0, it means that the bot has failed to create the comment, so we
-    // are free to create a new one here
-    if (id === 0) {
-      await octokit.rest.issues.createComment({
-        body: comment,
+      // Also update the check run status. As this is run under pytorch-bot,
+      // the check run will show up under that GitHub app
+      await octokit.rest.checks.create({
         owner: OWNER,
         repo: repo,
-        issue_number: pr_info.pr_number,
-      });
-    }
-    // Otherwise, update the existing comment
-    else {
-      await octokit.rest.issues.updateComment({
-        body: comment,
-        owner: OWNER,
-        repo: repo,
-        comment_id: id,
+        name: "Dr.CI",
+        head_sha: pr_info.head_sha,
+        status: "completed",
+        conclusion: "neutral",
+        output: {
+          title: "Dr.CI classification results",
+          // NB: the summary contains the classification result from Dr.CI,
+          // so that it can be queried elsewhere
+          summary: JSON.stringify(failures[pr_info.pr_number]),
```

Comment:
The character limitation is for this field where GH limits the max length of the summary to 65535 chars.  One possible tweak here is to remove the optional failure context here before JSON-ify to see if it helps.  But that could be done separately